# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/client/TextureClient.cpp
# Commit: 220d6288fe81
# Full Hash: 220d6288fe8156df13e9582af79702ae5ef1f237
# Author: Lee Salzman <lsalzman@mozilla.com>
# Date: 2022-12-09 21:45:26
# Regressor Bug: 1795768
# File Overlap Count: 1
# Description:
#   Bug 1795768 - Do not use Direct2D canvas if willReadFrequently is set. r=jrmuizel
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D164364
# ==============================================================================

diff -r 5d43ff0044fe -r 220d6288fe81 gfx/layers/client/TextureClient.cpp
--- a/gfx/layers/client/TextureClient.cpp	Fri Dec 09 17:03:03 2022 +0000
+++ b/gfx/layers/client/TextureClient.cpp	Fri Dec 09 17:03:03 2022 +0000
@@ -267,7 +267,7 @@
       (moz2DBackend == gfx::BackendType::DIRECT2D ||
        moz2DBackend == gfx::BackendType::DIRECT2D1_1) &&
       aSize.width <= maxTextureSize && aSize.height <= maxTextureSize &&
-      !(aAllocFlags & ALLOC_UPDATE_FROM_SURFACE)) {
+      !(aAllocFlags & (ALLOC_UPDATE_FROM_SURFACE | ALLOC_DO_NOT_ACCELERATE))) {
     return TextureType::D3D11;
   }
 #endif